### PR TITLE
refactor: Take Span in SetSeed

### PR DESCRIPTION
**Pull request by maflcko**
This makes calling code less verbose and less fragile. Also, by adding
the CKey::data() member function, it is now possible to call HexStr()
with a CKey object.

### CODE REVIEW

**Comment by sipa** [comment]
utACK fa93ef5a8aeae36304c792697a78af2d07fd9f41

File: src/test/key_io_tests.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@ BOOST_AUTO_TEST_CASE(key_io_valid_parse)
             privkey = DecodeSecret(exp_base58string);
             BOOST_CHECK_MESSAGE(privkey.IsValid(), "!IsValid:" + strTest);
             BOOST_CHECK_MESSAGE(privkey.IsCompressed() == isCompressed, "compressed mismatch:" + strTest);
-            BOOST_CHECK_MESSAGE(privkey.size() == exp_payload.size() && std::equal(privkey.begin(), privkey.end(), exp_payload.begin()), "key mismatch:" + strTest);
+            BOOST_CHECK_MESSAGE(Span<const uint8_t>{privkey} == Span<const uint8_t>{exp_payload}, "key mismatch:" + strTest);
```
Comment (theStack) [inline src/test/key_io_tests.cpp:49]: Note that C++20's `std::span` doesn't offer comparison operations (like `operator==`), i.e. we'll have to implement them ourselves as soon as we switch from `Span` to `std::span` (see e.g. https://brevzin.github.io/c++/2020/03/30/span-comparisons/ for anyone being interested in the topic).

**Review by theStack — APPROVED** [review]
Code-review ACK fa93ef5a8aeae36304c792697a78af2d07fd9f41

File: src/test/key_io_tests.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/test/key_io_tests.cpp:49]: Or as alternative, revert this hunk if this is the only use of `operator==` at the time

**Comment by laanwj** [comment]
Code review ACK fa93ef5a8aeae36304c792697a78af2d07fd9f41

**Comment by laanwj** [comment]
Closing manually because github doesn't seem to do it automatically.

**Pull request closed without merging**